dev/ToastNotifications/ToastNotificationManager.cpp

```diff
@@ -140,10 +140,58 @@ namespace winrt::Microsoft::Windows::ToastNotifications::implementation
 
         return static_cast<winrt::Microsoft::Windows::ToastNotifications::ToastNotificationSetting>(toastNotificationSetting);
     }
-    winrt::Microsoft::Windows::ToastNotifications::ToastNotificationHistory ToastNotificationManager::History()
+    winrt::Windows::Foundation::IAsyncAction ToastNotificationManager::RemoveWithIdentiferAsync(uint32_t /*toastIdentifier*/)
     {
         throw hresult_not_implemented();
     }
+    winrt::Windows::Foundation::IAsyncAction ToastNotificationManager::RemoveWithTagAsync(hstring /*tag*/)
+    {
+        throw hresult_not_implemented();
+    }
+    winrt::Windows::Foundation::IAsyncAction ToastNotificationManager::RemoveWithTagGroupAsync(hstring /*tag*/, hstring /*group*/)
+    {
+        throw hresult_not_implemented();
+    }
+    winrt::Windows::Foundation::IAsyncAction ToastNotificationManager::RemoveGroupAsync(hstring /*group*/)
+    {
+        throw hresult_not_implemented();
+    }
+    winrt::Windows::Foundation::IAsyncAction ToastNotificationManager::RemoveAllAsync()
+    {
+        co_await winrt::resume_background();
+
+        std::wstring appId{ RetrieveToastAppId() };
+        THROW_IF_FAILED(ToastNotifications_RemoveAllToastsForApp(appId.c_str()));
+
+        co_return;
```

Comment:
If this is not co-returning anything, do we still need to be this explicit?

Reply:
We don't. As long as there is at least one of: co_await, co_yield or co_return, the function is async.